pkg/cmd/pulumi/deployment_settings_config.go

```diff
@@ -180,63 +220,18 @@ func newDeploymentSettingsInitCmd() *cobra.Command {
 		Short:      "Initialize the stack's deployment.yaml file",
 		Long:       "",
 		Run: cmdutil.RunFunc(func(cmd *cobra.Command, args []string) error {
-			ctx := cmd.Context()
 			d, err := initializeDeploymentSettingsCmd(cmd, stack)
 			if err != nil {
 				return err
 			}
 
-			newStackDeployment := &workspace.ProjectStackDeployment{
-				DeploymentSettings: apitype.DeploymentSettings{},
-			}
-
-			err = configureGit(ctx, d.DisplayOptions, d.Backend, d.Stack, newStackDeployment, gitSSHPrivateKeyPath)
-			if err != nil {
-				return err
-			}
-
-			err = configureImageRepository(ctx, d.DisplayOptions, d.Backend, d.Stack, newStackDeployment)
-			if err != nil {
-				return err
-			}
-
-			err = configureAdvancedSettings(d.DisplayOptions, newStackDeployment)
-			if err != nil {
-				return err
-			}
-
-			var option string
-			if err := survey.AskOne(&survey.Select{
-				Message: "Do you want to configure an OpenID Connect integration?",
-				Options: []string{
-					optNo,
-					optOidcAws,
-					optOidcAzure,
-					optOidcGcp,
-				},
-				Default: optNo,
-			}, &option, surveyIcons(d.DisplayOptions.Color)); err != nil {
-				return errors.New("selection cancelled")
-			}
-
-			switch option {
-			case optOidcAws:
-				err = configureOidcAws(newStackDeployment)
-			case optOidcAzure:
-				err = configureOidcAzure(newStackDeployment)
-			case optOidcGcp:
-				err = configureOidcGCP(newStackDeployment)
-			}
-			if err != nil {
-				return err
-			}
-
-			err = saveProjectStackDeployment(newStackDeployment, d.Stack)
-			if err != nil {
-				return err
+			if d.Deployment != nil && !force {
+				return fmt.Errorf("Deployment settings already configured for stack %q. Rerun for a "+
+					"different stack by using --stack, update it by using the \"configure\" command or by "+
+					"editing the file manually; or use --force", d.Stack.Ref().FullyQualifiedName())
```

Comment:
I feel like here we should just use `Ref()`, because in theory the string instance should pick `FullyQualifiedName` if the user has set that flag? Could be wrong though.

Reply:
makes sense

Reply:
done